qiling/os/memory.py

```diff
@@ -63,6 +63,29 @@ def __read_string(self, addr: int) -> str:
     def __write_string(self, addr: int, s: str, encoding: str):
         self.write(addr, bytes(s, encoding) + b'\x00')
 
+    def __getitem__(self, key: Union[slice, int]) -> bytearray:
+        if isinstance(key, slice):
+            start = key.start
+            stop = key.stop
+            step = key.step
+
+            if step is not None and step != 1:
+                # step != 1 means we have to do copy, don't allow it.
+                raise IndexError("Only support slicing continous memory")
+
+            return self.ql.mem.read(start, stop - start)
+        elif isinstance(key, int):
+            return self.ql.mem.read(key, 1)
+        else:
+            raise KeyError("Wrong type of key")
+    
+    def __setitem__(self, key: Union[slice, int], value: Union[bytes, bytearray]):
+        if isinstance(key, int):
```

Comment:
Writing something like `ql.mem[0x1234] = b'hello world'` seems weird to me, as I would expect an integer key to represent a single byte, and not a starting location. That said, specifying a slice would not make sense either..

Reply:
now, it looks like `ql.mem[0x123] = 12`, which looks like `bytearray`

Reply:
`bytearray` would allow you to do something like `ba[3] = b'hello'`, and by that insert the data while expanding the array. That is not applicable to memory, so I honestly don't know what would make more sense here. Both options look wierd to me..

Reply:
Resolved by `ba[3:] = b"hello"`.